src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForClientParameterBuilder.cs

```diff
@@ -96,16 +97,33 @@ public AcquireTokenForClientParameterBuilder WithSendX5C(bool withSendX5C)
         /// <returns>The current instance of <see cref="AcquireTokenForClientParameterBuilder"/> to enable method chaining.</returns>
         public AcquireTokenForClientParameterBuilder WithMtlsProofOfPossession()
         {
-            if (ServiceBundle.Config.ClientCredential is not CertificateClientCredential certificateCredential)
+            if (ServiceBundle.Config.ClientCredential is CertificateClientCredential certCred)
             {
-                throw new MsalClientException(
-                    MsalError.MtlsCertificateNotProvided,
-                    MsalErrorMessage.MtlsCertificateNotProvidedMessage);
+                CommonParameters.AuthenticationOperation =
+                    new MtlsPopAuthenticationOperation(certCred.Certificate);
+                CommonParameters.MtlsCertificate = certCred.Certificate;
+            }
+            else if (ServiceBundle.Config.ClientCredential is ClientAssertionDelegateCredential assertCred)
+            {
+                X509Certificate2 cert = assertCred.PeekCertificate(ServiceBundle.Config.ClientId);
```

Comment:
This is not the correct place to for this logic. Instead it should be part of `ExecuteAsync` because executing the delegate is async.

Reply:
it is now in executeasync